packages/block-editor/src/hooks/duotone.js

```diff
@@ -97,7 +99,7 @@ ${ selector } {
 							values=".299 .587 .114 0 0
 							        .299 .587 .114 0 0
 							        .299 .587 .114 0 0
-							        0 0 0 1 0"
+							        .299 .587 .114 0 0"
```

Comment:
The problem with allowing transparency is this line and the corresponding one in [duotone.php](https://github.com/WordPress/gutenberg/pull/34130/files#diff-fd626ca219f3d805667d480eabb2d890a5a8d2e015b37f03cf8980edf1ae3d58L330-R373). This switches the alpha channel from transferring directly from the source image to calculating it based on the RGB values.

In order to maintain backwards compatibility, the default behavior will have to remain as using the source's alpha channel. So some sort of toggle would have to be added for the new functionality.